In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 20)

In [2]:
df = pd.read_csv("all_seasons.csv")

# Remove unnecessary index column
df.drop(columns="Unnamed: 0", inplace=True)

print(df.shape)
df.head()

(12844, 21)


,player_name,team_abbreviation,age,player_height,player_weight,college,country,draft_year,draft_round,draft_number,gp,pts,reb,ast,net_rating,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct,season
0,Randy Livingston,HOU,22.0,193.04,94.800728,Louisiana State,USA,1996,2,42,64,3.9,1.5,2.4,0.3,0.042,0.071,0.169,0.487,0.248,1996-97
1,Gaylon Nickerson,WAS,28.0,190.50,86.182480,Northwestern Oklahoma,USA,1994,2,34,4,3.8,1.3,0.3,8.9,0.030,0.111,0.174,0.497,0.043,1996-97
2,George Lynch,VAN,26.0,203.20,103.418976,North Carolina,USA,1993,1,12,41,8.3,6.4,1.9,-8.2,0.106,0.185,0.175,0.512,0.125,1996-97
3,George McCloud,LAL,30.0,203.20,102.058200,Florida State,USA,1989,1,7,64,10.2,2.8,1.7,-2.7,0.027,0.111,0.206,0.527,0.125,1996-97
4,George Zidek,DEN,23.0,213.36,119.748288,UCLA,USA,1995,1,22,52,2.8,1.7,0.3,-14.1,0.102,0.169,0.195,0.500,0.064,1996-97


In [3]:
missing = pd.DataFrame({
    "Missing Values": df.isnull().sum(),
    "Percentage": (df.isnull().mean() * 100).round(2)
})

missing.sort_values("Missing Values", ascending=False)

,Missing Values,Percentage
college,1854,14.43
player_name,0,0.00
pts,0,0.00
ast_pct,0,0.00
ts_pct,0,0.00
...,...,...
country,0,0.00
player_weight,0,0.00
player_height,0,0.00
age,0,0.00


In [4]:
duplicates = df[df.duplicated(subset=["player_name", "season"], keep=False)]

print("Duplicate Player-Season Records:", len(duplicates))

duplicates.sort_values(["player_name", "season"])

Duplicate Player-Season Records: 8


,player_name,team_abbreviation,age,player_height,player_weight,college,country,draft_year,draft_round,draft_number,gp,pts,reb,ast,net_rating,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct,season
7291,Chris Johnson,MIN,27.0,210.82,95.254320,Louisiana State,USA,Undrafted,Undrafted,Undrafted,30,3.9,2.0,0.3,-4.1,0.085,0.153,0.166,0.650,0.042,2012-13
7292,Chris Johnson,MEM,23.0,198.12,91.171992,Dayton,USA,Undrafted,Undrafted,Undrafted,8,3.6,1.4,0.3,18.1,0.051,0.073,0.124,0.560,0.025,2012-13
5163,Marcus Williams,LAC,21.0,200.66,93.893544,Arizona,USA,2007,2,33,11,0.9,1.1,0.3,-1.1,0.080,0.242,0.232,0.250,0.143,2007-08
5164,Marcus Williams,NJN,22.0,190.50,92.986360,Connecticut,USA,2006,1,22,53,5.9,1.9,2.6,-11.2,0.019,0.124,0.212,0.499,0.285,2007-08
5635,Marcus Williams,GSW,23.0,190.50,95.254320,Connecticut,USA,2006,1,22,9,1.3,0.4,1.4,-5.5,0.000,0.073,0.181,0.305,0.342,2008-09
5643,Marcus Williams,SAS,22.0,200.66,93.893544,Arizona,USA,2007,2,33,2,2.0,0.0,0.0,-52.3,0.000,0.000,0.240,1.000,0.000,2008-09
7962,Tony Mitchell,MIL,24.0,198.12,97.975872,Alabama,USA,Undrafted,Undrafted,Undrafted,3,2.0,0.3,0.3,3.1,0.167,0.000,0.242,0.600,0.200,2013-14
7963,Tony Mitchell,DET,22.0,203.20,106.594120,North Texas,USA,2013,2,37,21,1.0,1.2,0.1,2.6,0.174,0.177,0.132,0.540,0.040,2013-14


In [5]:
df["gp"].describe()

count    12844.000000
mean        51.154158
std         25.084904
min          1.000000
25%         31.000000
50%         57.000000
75%         73.000000
max         85.000000
Name: gp, dtype: float64

In [6]:
df[df["gp"] > 82].sort_values("gp", ascending=False)

,player_name,team_abbreviation,age,player_height,player_weight,college,country,draft_year,draft_round,draft_number,gp,pts,reb,ast,net_rating,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct,season
1595,Chris Gatling,DEN,32.0,208.28,99.790240,Old Dominion,USA,1991,1,16,85,11.9,5.9,0.8,-1.9,0.091,0.209,0.263,0.525,0.068,1999-00
3304,Shareef Abdur-Rahim,POR,27.0,205.74,111.130040,California,USA,1996,1,3,85,16.3,7.5,2.0,-2.2,0.082,0.183,0.248,0.557,0.126,2003-04
3377,Theo Ratliff,POR,31.0,208.28,106.594120,Wyoming,USA,1995,1,18,85,7.9,7.2,0.8,-0.4,0.088,0.168,0.136,0.521,0.043,2003-04
3752,Casey Jacobsen,NOH,24.0,198.12,97.522280,Stanford,USA,2002,1,22,84,6.5,2.0,1.3,-0.4,0.023,0.081,0.144,0.555,0.102,2004-05
382,Michael Finley,DAL,24.0,200.66,97.522280,Wisconsin,USA,1995,1,21,83,15.0,4.5,2.7,-6.5,0.036,0.120,0.221,0.530,0.147,1996-97
419,A.C. Green,DAL,33.0,205.74,102.058200,Oregon State,USA,1985,1,23,83,7.2,7.9,0.8,-8.0,0.100,0.207,0.119,0.523,0.045,1996-97
420,Aaron McKie,DET,24.0,195.58,94.800728,Temple,USA,1994,1,17,83,5.2,2.7,1.9,5.2,0.031,0.129,0.147,0.524,0.163,1996-97
1874,Mark Jackson,NYK,36.0,190.50,92.986360,St. John's (NY),USA,1987,1,18,83,7.6,3.7,8.0,0.4,0.028,0.107,0.143,0.505,0.396,2000-01
2381,Jalen Rose,CHI,29.0,203.20,98.429464,Michigan,USA,1994,1,13,83,20.4,4.5,4.3,-2.1,0.016,0.117,0.260,0.528,0.203,2001-02
2620,Bo Outlaw,PHX,31.0,203.20,95.254320,Houston,USA,Undrafted,Undrafted,Undrafted,83,4.5,4.4,1.5,2.1,0.088,0.128,0.104,0.547,0.097,2001-02


In [7]:
df["net_rating"].describe()

count    12844.000000
mean        -2.226339
std         12.665124
min       -250.000000
25%         -6.400000
50%         -1.300000
75%          3.200000
max        300.000000
Name: net_rating, dtype: float64

In [8]:
df.sort_values("net_rating").head(10)

,player_name,team_abbreviation,age,player_height,player_weight,college,country,draft_year,draft_round,draft_number,gp,pts,reb,ast,net_rating,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct,season
12259,Sam Dekker,TOR,28.0,203.20,99.336648,Wisconsin,USA,2015,1,18,1,0.0,0.0,0.0,-250.0,0.0,0.00,0.000,0.000,0.0,2021-22
953,Gheorghe Muresan,NJN,28.0,231.14,137.438376,NaN,USA,1993,2,30,1,0.0,0.0,0.0,-200.0,0.0,0.00,1.000,0.000,0.0,1998-99
1553,Jason Miskiri,CHH,24.0,187.96,83.914520,George Mason,USA,Undrafted,Undrafted,Undrafted,1,0.0,0.0,1.0,-158.3,0.0,0.00,0.125,0.000,0.5,1999-00
9728,Tyler Lydon,DEN,22.0,205.74,99.790240,Syracuse,USA,2017,1,24,1,0.0,0.0,0.0,-150.0,0.0,0.00,0.000,0.000,0.0,2017-18
11475,Will Magnay,NOP,23.0,208.28,106.140528,NaN,Australia,Undrafted,Undrafted,Undrafted,1,0.0,0.0,0.0,-148.2,0.0,0.00,0.200,0.000,0.0,2020-21
28,Elmore Spencer,SEA,27.0,213.36,122.469840,Nevada-Las Vegas,USA,1992,1,25,1,0.0,0.0,0.0,-147.5,0.0,0.00,0.126,0.000,0.0,1996-97
6269,Hassan Whiteside,SAC,22.0,213.36,106.594120,Marshall,USA,2010,2,33,1,0.0,0.0,0.0,-144.9,0.0,0.00,0.000,0.000,0.0,2010-11
4620,Mile Ilic,NJN,23.0,215.90,115.665960,NaN,USA,2005,2,43,5,0.0,0.2,0.0,-143.1,0.0,0.25,0.750,0.000,0.0,2006-07
4227,Matt Walsh,MIA,23.0,198.12,99.790240,Florida,USA,Undrafted,Undrafted,Undrafted,2,1.0,0.0,0.0,-140.5,0.0,0.00,0.324,0.532,0.0,2005-06
9829,Edmond Sumner,IND,22.0,195.58,85.275296,Xavier,USA,2017,2,52,1,2.0,1.0,0.0,-120.0,0.0,0.50,0.200,1.000,0.0,2017-18


In [9]:
df.sort_values("net_rating", ascending=False).head(10)

,player_name,team_abbreviation,age,player_height,player_weight,college,country,draft_year,draft_round,draft_number,gp,pts,reb,ast,net_rating,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct,season
127,Bruce Bowen,MIA,26.0,200.66,90.718400,Cal State-Fullerton,USA,Undrafted,Undrafted,Undrafted,1,0.0,0.0,0.0,300.0,0.000,0.000,0.000,0.000,0.000,1996-97
9608,Naz Mitrou-Long,UTA,24.0,193.04,98.883056,Iowa State,Canada,Undrafted,Undrafted,Undrafted,1,3.0,0.0,0.0,250.0,0.000,0.000,0.500,1.500,0.000,2017-18
10692,J.P. Macura,CLE,25.0,193.04,92.079176,Xavier,USA,2018,Undrafted,Undrafted,1,0.0,0.0,0.0,150.0,0.000,0.000,0.000,0.000,0.000,2019-20
11963,Ahmad Caver,IND,25.0,187.96,78.925008,Old Dominion,USA,2019,Undrafted,Undrafted,1,2.0,0.0,0.0,133.3,0.000,0.000,0.500,1.000,0.000,2021-22
8571,Jerrelle Benimon,DEN,23.0,203.00,111.000000,NaN,USA,Undrafted,Undrafted,Undrafted,2,0.0,1.5,0.0,120.0,1.000,0.400,0.167,0.000,0.000,2014-15
6802,Hamady Ndiaye,WAS,25.0,213.36,106.594120,Rutgers,Senegal,2010,2,56,3,0.0,0.0,0.0,114.3,0.000,0.000,0.111,0.000,0.000,2011-12
11069,Max Strus,CHI,24.0,195.58,97.522280,DePaul,USA,Undrafted,Undrafted,Undrafted,2,2.5,0.5,0.0,105.8,0.167,0.000,0.158,0.727,0.000,2019-20
11977,DeJon Jarreau,IND,24.0,195.58,83.914520,Houston,USA,Undrafted,Undrafted,Undrafted,1,0.0,0.0,0.0,100.0,0.000,0.000,0.000,0.000,0.000,2021-22
12435,Stanley Umude,DET,24.0,198.12,95.254320,Arkansas,USA,Undrafted,Undrafted,Undrafted,1,2.0,0.0,0.0,95.0,0.000,0.000,0.500,0.532,0.000,2022-23
3312,Ryan Humphrey,MEM,24.0,203.20,106.594120,Notre Dame,USA,2002,1,19,2,1.0,1.5,0.5,94.8,0.286,0.111,0.226,0.250,0.083,2003-04


In [10]:
df["ts_pct"].describe()

count    12844.000000
mean         0.513138
std          0.101724
min          0.000000
25%          0.482000
50%          0.525000
75%          0.563000
max          1.500000
Name: ts_pct, dtype: float64

In [11]:
df[df["ts_pct"] > 1]

,player_name,team_abbreviation,age,player_height,player_weight,college,country,draft_year,draft_round,draft_number,gp,pts,reb,ast,net_rating,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct,season
551,Tom Chambers,PHI,39.0,208.28,99.790240,Utah,USA,1981,1,8,1,6.0,2.0,0.0,-17.8,0.000,0.167,0.157,1.042,0.000,1997-98
1151,Tyson Wheeler,DEN,23.0,177.80,74.842680,Rhode Island,USA,1998,2,47,1,4.0,0.0,2.0,62.2,0.000,0.000,0.320,1.064,0.667,1998-99
1261,Mikki Moore,DET,23.0,213.36,104.326160,Nebraska,USA,Undrafted,Undrafted,Undrafted,2,2.0,0.5,0.0,89.7,0.000,0.167,0.158,1.064,0.000,1998-99
3879,Jackie Butler,NYK,20.0,208.28,113.398000,Southern Illinois,USA,Undrafted,Undrafted,Undrafted,3,3.3,0.0,0.0,5.1,0.000,0.000,0.595,1.025,0.000,2004-05
3939,Donnell Harvey,NJN,24.0,203.20,99.790240,Florida,USA,2000,1,22,3,2.7,2.3,0.3,21.4,0.200,0.455,0.149,1.064,0.125,2004-05
4479,Dajuan Wagner,GSW,24.0,190.50,90.718400,Memphis,USA,2002,1,6,1,4.0,0.0,1.0,4.9,0.000,0.000,0.207,1.064,0.200,2006-07
5196,Jerome James,NYK,32.0,215.90,129.273720,Florida A&M,USA,1998,2,36,2,2.0,1.5,0.0,52.2,0.000,0.600,0.190,1.064,0.000,2007-08
5787,Ryan Bowen,OKC,34.0,205.74,98.883056,Iowa,USA,1998,2,55,1,4.0,2.0,0.0,69.2,0.111,0.100,0.083,1.064,0.000,2009-10
8246,Tyrus Thomas,MEM,28.0,208.28,102.058200,Louisiana State,USA,2006,1,4,2,2.0,0.0,0.5,3.7,0.000,0.000,0.137,1.064,0.200,2014-15
9608,Naz Mitrou-Long,UTA,24.0,193.04,98.883056,Iowa State,Canada,Undrafted,Undrafted,Undrafted,1,3.0,0.0,0.0,250.0,0.000,0.000,0.500,1.500,0.000,2017-18


In [12]:
percentage_columns = [
    "oreb_pct",
    "dreb_pct",
    "usg_pct",
    "ts_pct",
    "ast_pct"
]

df[percentage_columns].describe()

,oreb_pct,dreb_pct,usg_pct,ts_pct,ast_pct
count,12844.000000,12844.000000,12844.000000,12844.000000,12844.000000
mean,0.054073,0.140646,0.184641,0.513138,0.131595
std,0.043335,0.062513,0.053545,0.101724,0.094172
min,0.000000,0.000000,0.000000,0.000000,0.000000
25%,0.021000,0.096000,0.149000,0.482000,0.066000
50%,0.040000,0.130500,0.181000,0.525000,0.103000
75%,0.083000,0.179000,0.217000,0.563000,0.179000
max,1.000000,1.000000,1.000000,1.500000,1.000000


In [13]:
df["draft_year"].value_counts().head(10)

draft_year
Undrafted    2358
1998          454
2003          435
2005          427
2008          422
1996          406
2001          404
2009          395
2011          374
2004          371
Name: count, dtype: int64

In [14]:
df["draft_round"].value_counts()

draft_round
1            7351
2            3032
Undrafted    2411
3              20
4              12
0               6
6               5
7               5
8               2
Name: count, dtype: int64

In [15]:
df["draft_number"].value_counts().head(15)

draft_number
Undrafted    2414
1             361
5             356
4             345
3             339
2             327
9             325
10            311
7             309
8             291
13            271
11            263
12            243
24            239
15            238
Name: count, dtype: int64

In [16]:
sorted(df["team_abbreviation"].unique())

['ATL',
 'BKN',
 'BOS',
 'CHA',
 'CHH',
 'CHI',
 'CLE',
 'DAL',
 'DEN',
 'DET',
 'GSW',
 'HOU',
 'IND',
 'LAC',
 'LAL',
 'MEM',
 'MIA',
 'MIL',
 'MIN',
 'NJN',
 'NOH',
 'NOK',
 'NOP',
 'NYK',
 'OKC',
 'ORL',
 'PHI',
 'PHX',
 'POR',
 'SAC',
 'SAS',
 'SEA',
 'TOR',
 'UTA',
 'VAN',
 'WAS']

In [17]:
sorted(df["season"].unique())

['1996-97',
 '1997-98',
 '1998-99',
 '1999-00',
 '2000-01',
 '2001-02',
 '2002-03',
 '2003-04',
 '2004-05',
 '2005-06',
 '2006-07',
 '2007-08',
 '2008-09',
 '2009-10',
 '2010-11',
 '2011-12',
 '2012-13',
 '2013-14',
 '2014-15',
 '2015-16',
 '2016-17',
 '2017-18',
 '2018-19',
 '2019-20',
 '2020-21',
 '2021-22',
 '2022-23']

In [18]:
career_length = (
    df.groupby("player_name")["season"]
      .nunique()
      .sort_values(ascending=False)
)

career_length.head(20)

player_name
Vince Carter       22
Dirk Nowitzki      21
Udonis Haslem      20
Kobe Bryant        20
Jamal Crawford     20
Kevin Garnett      20
LeBron James       20
Tyson Chandler     19
Tim Duncan         19
Carmelo Anthony    19
Jason Terry        19
Andre Iguodala     19
Paul Pierce        19
Nazr Mohammed      18
Steve Nash         18
Jermaine O'Neal    18
Dwight Howard      18
Joe Johnson        18
Derek Fisher       18
Ray Allen          18
Name: season, dtype: int64

In [19]:
teams = (
    df.groupby("player_name")["team_abbreviation"]
      .nunique()
      .sort_values(ascending=False)
)

teams.head(20)

player_name
Jeff Green          11
Mike James          11
Jim Jackson         10
Randy Livingston     9
James Johnson        9
Rasual Butler        9
Jamal Crawford       9
Mikki Moore          9
Tyson Chandler       9
Keith Bogans         9
Anthony Tolliver     9
Jarrett Jack         9
Robin Lopez          9
Matt Barnes          8
Earl Boykins         8
Garrett Temple       8
Caron Butler         8
Ed Davis             8
Jerry Stackhouse     8
Shaun Livingston     8
Name: team_abbreviation, dtype: int64

In [20]:
print("========== DATA QUALITY AUDIT ==========\n")

print(f"Rows: {len(df)}")
print(f"Columns: {len(df.columns)}")

print(f"\nDuplicate Rows: {df.duplicated().sum()}")

print(f"Missing Colleges: {df['college'].isnull().sum()}")

print(f"\nUnique Players: {df['player_name'].nunique()}")

print(f"Seasons Covered: {df['season'].nunique()}")

print(f"Teams: {df['team_abbreviation'].nunique()}")

print(f"\nPlayers with GP > 82: {(df['gp'] > 82).sum()}")

print(f"Players with TS% > 1: {(df['ts_pct'] > 1).sum()}")

print(f"Net Rating < -50: {(df['net_rating'] < -50).sum()}")

print(f"Net Rating > 50: {(df['net_rating'] > 50).sum()}")

========== DATA QUALITY AUDIT ==========

Rows: 12844
Columns: 21

Duplicate Rows: 0
Missing Colleges: 1854

Unique Players: 2551
Seasons Covered: 27
Teams: 36

Players with GP > 82: 15
Players with TS% > 1: 15
Net Rating < -50: 95
Net Rating > 50: 35


In [21]:
print("="*60)
print("NBA PLAYER STATS DATASET OVERVIEW")
print("="*60)

print(f"Rows               : {df.shape[0]}")
print(f"Columns            : {df.shape[1]}")
print(f"Unique Players     : {df['player_name'].nunique()}")
print(f"NBA Seasons        : {df['season'].nunique()}")
print(f"Teams              : {df['team_abbreviation'].nunique()}")
print(f"Countries          : {df['country'].nunique()}")
print(f"Colleges           : {df['college'].nunique()}")

NBA PLAYER STATS DATASET OVERVIEW
Rows               : 12844
Columns            : 21
Unique Players     : 2551
NBA Seasons        : 27
Teams              : 36
Countries          : 82
Colleges           : 356


In [22]:
numerical_cols = df.select_dtypes(include=["int64", "float64"]).columns.tolist()
categorical_cols = df.select_dtypes(include="object").columns.tolist()

print("Numerical Features")
print(numerical_cols)

print("\nCategorical Features")
print(categorical_cols)

Numerical Features
['age', 'player_height', 'player_weight', 'gp', 'pts', 'reb', 'ast', 'net_rating', 'oreb_pct', 'dreb_pct', 'usg_pct', 'ts_pct', 'ast_pct']

Categorical Features
['player_name', 'team_abbreviation', 'college', 'country', 'draft_year', 'draft_round', 'draft_number', 'season']


In [23]:
feature_summary = pd.DataFrame({
    "Data Type": df.dtypes,
    "Missing Values": df.isnull().sum(),
    "Unique Values": df.nunique()
})

feature_summary

,Data Type,Missing Values,Unique Values
player_name,object,0,2551
team_abbreviation,object,0,36
age,float64,0,27
player_height,float64,0,30
player_weight,float64,0,157
...,...,...,...
dreb_pct,float64,0,352
usg_pct,float64,0,348
ts_pct,float64,0,563
ast_pct,float64,0,483


In [24]:
print("Negative Points :", (df["pts"] < 0).sum())
print("Negative Rebounds :", (df["reb"] < 0).sum())
print("Negative Assists :", (df["ast"] < 0).sum())

print("Age < 18 :", (df["age"] < 18).sum())

print("Height < 160 cm :", (df["player_height"] < 160).sum())
print("Weight < 60 kg :", (df["player_weight"] < 60).sum())

Negative Points : 0
Negative Rebounds : 0
Negative Assists : 0
Age < 18 : 0
Height < 160 cm : 0
Weight < 60 kg : 0


## Dataset Description

The dataset contains NBA player season statistics from the **1996–97** season through the **2022–23** season.

Each row represents one player's performance during a single NBA season and contains information about:

- Player information (name, age, height, weight)
- Team information
- Draft details
- College and country
- Performance statistics (points, rebounds, assists, games played)
- Advanced basketball metrics such as Net Rating, True Shooting Percentage (TS%), Usage Percentage (USG%), Offensive Rebound Percentage (OREB%), Defensive Rebound Percentage (DREB%), and Assist Percentage (AST%).

The dataset consists of **12,844 records** and **21 features**.

## Dataset Dimensions

- Total Rows: **12,844**
- Total Columns: **21**
- Unique Players: **2,551**
- Seasons Covered: **27**
- Teams Represented: **36**
- Countries Represented: **82**
- Colleges Represented: **356**

## Feature Types

The dataset contains two main types of variables.

### Numerical Features
These include continuous and discrete player statistics such as:

- Age
- Height
- Weight
- Games Played (GP)
- Points (PTS)
- Rebounds (REB)
- Assists (AST)
- Net Rating
- Offensive Rebound %
- Defensive Rebound %
- Usage %
- True Shooting %
- Assist %

### Categorical Features
These describe player identity and background:

- Player Name
- Team Abbreviation
- College
- Country
- Draft Year
- Draft Round
- Draft Number
- Season


## Missing Values Analysis

Only one feature contains missing values:

| Feature | Missing Values | Percentage |
|---------|---------------:|-----------:|
| College | 1,854 | 14.43% |

The missing college values are expected because many NBA players entered the league directly from high school or international leagues, making these missing values acceptable rather than data errors.


## Duplicate Records

No completely duplicated rows were found.

However, eight records share the same player name and season. These are not actual duplicates because they belong to different players with identical names (for example, two different players named Marcus Williams).

Therefore, no duplicate records were removed.

## Data Quality Checks

Several validation checks were performed on the dataset.

The following were verified:

- No negative values for points, rebounds, or assists.
- No players younger than 18 years.
- No unrealistic player heights or weights.
- No completely duplicated records.

Some extreme values were identified:

- Players with more than 82 games played.
- TS% values greater than 1.
- Extremely high and low Net Rating values.

These values mostly belong to players with very few games played, causing advanced statistics to become unstable. They were retained because they represent valid observations rather than data entry errors.


## Observations

Several interesting observations were identified during the exploratory analysis:

- Vince Carter appears in the largest number of seasons (22).
- Jeff Green and Mike James played for the highest number of NBA teams (11 each).
- Approximately 14% of players have missing college information.
- Undrafted players form the largest draft category.
- Most player statistics fall within realistic NBA ranges.
- Advanced statistics such as Net Rating and True Shooting Percentage contain extreme values caused by players with very limited playing time.

## Conclusion

The dataset is generally clean and suitable for analysis.

Apart from missing values in the College column and a few expected statistical outliers, no major data quality issues were found.

The dataset is ready for exploratory data analysis, visualization, feature engineering, and predictive modeling.